frame/nomination-pools/src/lib.rs

```diff
@@ -582,25 +583,237 @@ pub struct PoolRoles<AccountId> {
+	fn throttling(&self, to: &Perbill) -> bool {
+		if let Some(t) = self.change_rate.as_ref() {
+			let commission_as_percent =
+				self.current.as_ref().map(|(x, _)| *x).unwrap_or(Perbill::zero());
+
+			// factor previously updated block into whether user is throttled.
+			if self.last_updated.map_or(
+				// if no `last_updated` is set, throttled if the attempted increase in
+				// commission is greater than `max_increase`.
+				(*to).saturating_sub(commission_as_percent) > t.max_increase,
+				|p| {
+					// `min_delay` blocks must have been surpassed since last update.
+					if <frame_system::Pallet<T>>::block_number().saturating_sub(p) < t.min_delay {
+						return true
+					}
+					// ensure the `max_increase` durations surpassed since the previous commission
+					// update allow the attempted commission increase.
+					//
+					// the attempted increase in commission relative to the current commission.
+					let attempted_increase = (*to).saturating_sub(commission_as_percent);
+
+					// the total durations passed since the last commission update.
+					let blocks_passed = <frame_system::Pallet<T>>::block_number().saturating_sub(p);
+
+					// calculate intervals passed since last `last_updated`
+					let intervals_passed = if blocks_passed == Zero::zero() {
+						Zero::zero()
+					} else {
+						blocks_passed.div(t.min_delay).saturated_into::<u32>()
+					};
+
+					// the maximum allowed increase, where the current `max_increase` Perbill is
+					// converted into a u32 by multiplying itself with 100_u32 and multiplied by
+					// durations passed, before being converted back into a Perbill.
+					let max_allowed_increase =
+						Perbill::from_percent((t.max_increase * 100_u32) * intervals_passed);
+
+					// throttled (true) if attempted increase is greater than the maximum allowed
+					// increase, as a percentage.
+					attempted_increase > max_allowed_increase
+				},
+			) {
+				return true
+			}
+		}
+		false
+	}
+
+	/// Set the pool's commission.
+	///
+	/// Update commission based on `current`. If a `None` current value is supplied, allow the
+	/// commission to be removed in all cases, without any throttling restrictions. If `change_rate`
+	/// is present, record the current block  as the previously updated commission. If the supplied
+	/// commission is zero, `None` will be  inserted and `payee` will be ignored.
+	fn maybe_update_current(
+		&mut self,
+		current: &Option<(Perbill, T::AccountId)>,
+	) -> DispatchResult {
+		if current.is_none() {
+			self.current = None;
+		} else {
+			let (commission, payee) = current.as_ref().ok_or(Error::<T>::NoCommissionSet)?;
+			ensure!(!self.throttling(&commission), Error::<T>::CommissionChangeThrottled);
+			ensure!(
+				self.max.map_or(true, |m| commission <= &m),
+				Error::<T>::CommissionExceedsMaximum
+			);
+			self.current = Some((*commission, payee.clone()));
+		}
+		let _ = self.register_update(<frame_system::Pallet<T>>::block_number());
```

Comment:
@Ank4n `throttle.previous_set_at` has been moved to a standalone `last_updated` field, and is updated every time commission is set.